toolchain/base/kind_switch.h

```diff
@@ -50,65 +50,116 @@ template <typename... Ts>
 template <typename T>
 concept IsStdVariant = IsStdVariantValue<std::decay_t<T>>;
 
-template <typename... Ts>
-struct TypePack {};
-
-template <typename T>
-struct StdVariantTypePackValue;
+#define CARBON_INTERNAL_IDENTIFIER(name) T##name
+// Turns a list of numbers into a list `T0, T1, ...`.
+#define CARBON_INTERNAL_IDENTIFIERS(...)                             \
```

Comment:
I think these macro names are a bit too general since they're scoped over the whole toolchain. Maybe include something about kind switch here?